src/providers/info.rs

```diff
@@ -165,20 +169,22 @@ impl<C: Cache> InfoProvider for DefaultInfoProvider<'_, C> {
         if contract_info.display_name.trim().is_empty() {
             bail!("No display name")
         } else {
-            Ok(AddressInfo {
-                name: contract_info.display_name.to_owned(),
-                logo_uri: contract_info.logo_uri.to_owned(),
+            Ok(AddressEx {
+                value: address.to_owned(),
+                name: Some(contract_info.display_name.to_owned()),
+                logo_url: contract_info.logo_uri.to_owned(),
             })
         }
     }
 
-    async fn full_address_info_search(&self, address: &str) -> ApiResult<AddressInfo> {
+    async fn address_ex_from_any_source(&self, address: &str) -> ApiResult<AddressEx> {
         self.token_info(&address)
-            .map_ok(|it| AddressInfo {
-                name: it.name,
-                logo_uri: it.logo_uri,
+            .map_ok(|it| AddressEx {
+                value: address.to_owned(),
+                name: Some(it.name),
+                logo_url: it.logo_uri,
```

Comment:
the `uri/url` renaming will be addressed in a different issue, right?

Reply:
Yes